sdk/core/core-rest-pipeline/src/policies/decompressResponsePolicy.ts

```diff
@@ -17,7 +17,10 @@ export function decompressResponsePolicy(): PipelinePolicy {
   return {
     name: decompressResponsePolicyName,
     async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
-      request.headers.set("Accept-Encoding", "gzip,deflate");
+      // HEAD requests have no body
+      if (request.method !== "HEAD") {
```

Comment:
Are there other places where a similar check could be needed? I just looked and I am not sure tbh.

Reply:
I looked at all the other supported HTTP methods and none of them had this semantic (even though ones that don't always return a body may in some cases)